modules/service/src/main/resources/db/migration/V0361__chron_program_updates.sql

```diff
@@ -0,0 +1,144 @@
+  mod_pts_execution bool := NEW.c_pts_execution IS DISTINCT FROM OLD.c_pts_execution;
+BEGIN
+
+  INSERT INTO t_chron_program_update AS chron (
+    c_operation,
+    c_program_id          ,
+    c_mod_program_id      ,
+    c_mod_existence       ,
+    c_mod_pi_user_id      ,
+    c_mod_pi_user_type    ,
+    c_mod_name            ,
+    c_mod_pts_pi          ,
+    c_mod_pts_uncharged   ,
+    c_mod_pts_execution   ,
+    c_new_program_id      ,
+    c_new_existence       ,
+    c_new_pi_user_id      ,
+    c_new_pi_user_type    ,
+    c_new_name            ,
+    c_new_pts_pi          ,
+    c_new_pts_uncharged   ,
+    c_new_pts_execution
+  ) VALUES (
+    TG_OP::e_tg_op,
+    coalesce(OLD.c_program_id, NEW.c_program_id),
+    mod_program_id,   
+    mod_existence,    
+    mod_pi_user_id,   
+    mod_pi_user_type, 
+    mod_name,         
+    mod_pts_pi,       
+    mod_pts_uncharged,
+    mod_pts_execution,
+    CASE WHEN mod_program_id    = true THEN NEW.c_program_id    END,
+    CASE WHEN mod_existence     = true THEN NEW.c_existence     END,
+    CASE WHEN mod_pi_user_id    = true THEN NEW.c_pi_user_id    END,
+    CASE WHEN mod_pi_user_type  = true THEN NEW.c_pi_user_type  END,
+    CASE WHEN mod_name          = true THEN NEW.c_name          END,
+    CASE WHEN mod_pts_pi        = true THEN NEW.c_pts_pi        END,
+    CASE WHEN mod_pts_uncharged = true THEN NEW.c_pts_uncharged END,
+    CASE WHEN mod_pts_execution = true THEN NEW.c_pts_execution END
+  ) ON CONFLICT ON CONSTRAINT t_chron_program_update_unique DO UPDATE SET
+    c_mod_existence     = chron.c_mod_existence     OR mod_existence,    
+    c_mod_pi_user_id    = chron.c_mod_pi_user_id    OR mod_pi_user_id,   
+    c_mod_pi_user_type  = chron.c_mod_pi_user_type  OR mod_pi_user_type, 
+    c_mod_name          = chron.c_mod_name          OR mod_name,         
+    c_mod_pts_pi        = chron.c_mod_pts_pi        OR mod_pts_pi,       
+    c_mod_pts_uncharged = chron.c_mod_pts_uncharged OR mod_pts_uncharged,
+    c_mod_pts_execution = chron.c_mod_pts_execution OR mod_pts_execution,
+    c_new_program_id    = CASE WHEN chron.c_mod_program_id    OR mod_program_id    THEN NEW.c_program_id    END,
+    c_new_existence     = CASE WHEN chron.c_mod_existence     OR mod_existence     THEN NEW.c_existence     END,
+    c_new_pi_user_id    = CASE WHEN chron.c_mod_pi_user_id    OR mod_pi_user_id    THEN NEW.c_pi_user_id    END,
+    c_new_pi_user_type  = CASE WHEN chron.c_mod_pi_user_type  OR mod_pi_user_type  THEN NEW.c_pi_user_type  END,
+    c_new_name          = CASE WHEN chron.c_mod_name          OR mod_name          THEN NEW.c_name          END,
+    c_new_pts_pi        = CASE WHEN chron.c_mod_pts_pi        OR mod_pts_pi        THEN NEW.c_pts_pi        END,
+    c_new_pts_uncharged = CASE WHEN chron.c_mod_pts_uncharged OR mod_pts_uncharged THEN NEW.c_pts_uncharged END,
+    c_new_pts_execution = CASE WHEN chron.c_mod_pts_execution OR mod_pts_execution THEN NEW.c_pts_execution END;
```

Comment:
This is likely going to change soon, which raises the point that the `chron` tables will also have to be migrated whenever the associated table itself is migrated.

Reply:
Yes, the Chronicle tables will need to be updated as the model changes. I don't have a sense for how hard that will be.